qgis/examples/elevation_tile_downloader/core/task.py

```diff
@@ -1,117 +1,122 @@
-                    reply: QgsNetworkReplyContent = self.nam.blockingGet(request_)
-                    length = int(reply.rawHeader(bytes('Content-Length'.encode())))
-                    stream = BytesIO(reply.content())
-                    with gzip.GzipFile(fileobj=stream, mode='rb') as gz:
-                        with open(filepath, 'wb') as f:
-                            processed = 0
-                            chunk_size = 1024 ** 2
-                            while True:
-                                chunk = gz.read(chunk_size)
-                                if not chunk:
-                                    break
-                                f.write(chunk)
-                                processed += chunk_size
-                                self.setProgress((processed / length) * 100)  # goes to 90-something percent real fast and then stays there for another couple of seconds or so...
-                else:
-                    QgsMessageLog.logMessage(
-                        message=f"Already downloaded tile {url.url()}", level=Qgis.Info
-                    )
-                self.tiles.append(filepath)
+            for i, bbox in enumerate(self.grid):
                 if self.isCanceled():
                     return False
+
+                # Get the filepath and create the output directory (if necessary)
+                x, y = (
+                    bbox.boundingBox().xMinimum(),
+                    bbox.boundingBox().yMinimum(),
+                )
+                hemisphere = "S" if y < 0 else "N"
+                dir_name = "%s%02d" % (hemisphere, abs(y))
+                directory = self.output_dir.joinpath(dir_name)
+                directory.mkdir(exist_ok=True)
+                tile_name = "%s%02d%s%03d.hgt" % (
+                    hemisphere,
+                    abs(y),
+                    "W" if x < 0 else "E",
+                    abs(x),
+                )
+                filepath = self.output_dir.joinpath(dir_name, tile_name)
+                self.tiles.append(filepath)
+                if filepath.is_file():
+                    QgsMessageLog.logMessage(
+                        message=f"Already downloaded tile {tile_name}",
+                        level=Qgis.Info,
+                    )
+                    continue
+
+                # get the new tile if it didn't exist
+                url = QUrl(
+                    f"http://s3.amazonaws.com/elevation-tiles-prod/skadi/{dir_name}/{tile_name}.gz"
+                )
+                request = QNetworkRequest(url)
+                reply: QgsNetworkReplyContent = nam.blockingGet(request)
+                # turn the reply into a file object and write the decompressed file to disk
+                with gzip.GzipFile(
+                    fileobj=BytesIO(reply.content()), mode="rb"
+                ) as gz:
+                    with open(filepath, "wb") as f:
+                        f.write(gz.read())
```

Comment:
I was asking you to do it the way you tried it (progressing per file download), but thinking about it again, you're right, it doesn't make much sense, since those tiles are so tiny. progress is now incremented per tile download.